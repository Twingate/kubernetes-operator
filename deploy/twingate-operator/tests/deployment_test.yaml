suite: Deployment
templates:
  - deployment.yaml
tests:
  - it: should use `existingAPIKeySecret`
    set:
      twingateOperator:
        network: "<network slug>"
        existingAPIKeySecret:
          name: api-key-secret
          key: TWINGATE_API_KEY
        remoteNetworkId: "<remote network id>"
    asserts:
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: TWINGATE_API_KEY
            valueFrom:
              secretKeyRef:
                name: api-key-secret
                key: TWINGATE_API_KEY
  - it: should use `existingRemoteNetworkIdSecret`
    set:
      twingateOperator:
        apiKey: "<api key>"
        network: "<network slug>"
        existingRemoteNetworkIdSecret:
          name: remote-network-id-secret
          key: TWINGATE_REMOTE_NETWORK_ID
    asserts:
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: TWINGATE_REMOTE_NETWORK_ID
            valueFrom:
              secretKeyRef:
                name: remote-network-id-secret
                key: TWINGATE_REMOTE_NETWORK_ID
  - it: should use `remoteNetworkName`
    set:
      twingateOperator:
        apiKey: "<api key>"
        network: "<network slug>"
        remoteNetworkName: "<remote network name>"
    asserts:
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: TWINGATE_REMOTE_NETWORK_NAME
            value: "<remote network name>"
  - it: should require `network`
    set:
      twingateOperator:
        apiKey: "<api key>"
        remoteNetworkId: "<remote network id>"
    asserts:
      - failedTemplate:
          errorPattern: "network is required"
  - it: should require `apiKey`
    set:
      twingateOperator:
        network: "<network slug>"
        remoteNetworkId: "<remote network id>"
    asserts:
      - failedTemplate:
          errorPattern: "apiKey is required"
  - it: should require Remote Network
    set:
      twingateOperator:
        apiKey: "<api key>"
        network: "<network slug>"
    asserts:
      - failedTemplate:
          errorPattern: "remoteNetworkId is required"
  - it: should not allow multiple remote network definitions
    set:
      twingateOperator:
        apiKey: "<api key>"
        network: "<network slug>"
        remoteNetworkId: "<remote network id>"
        existingRemoteNetworkIdSecret:
          name: my-secret
          key: MY_TWINGATE_REMOTE_NETWORK_ID
    asserts:
      - failedTemplate:
          errorPattern: "twingateOperator: Must validate one and only one schema (oneOf)"
