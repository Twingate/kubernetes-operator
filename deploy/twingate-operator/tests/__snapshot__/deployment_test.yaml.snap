should set optional values:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: twingate-operator
        app.kubernetes.io/version: latest
        helm.sh/chart: twingate-operator-0.1.13
      name: RELEASE-NAME-twingate-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: twingate-operator
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            foo: bar
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: twingate-operator
            foo: bar
        spec:
          affinity:
            foo: bar
          containers:
            - command:
                - kopf
                - run
                - ./main.py
                - -A
                - --standalone
                - --liveness=http://0.0.0.0:8080/healthz
                - --log-format=plain
                - --verbose
              env:
                - name: TWINGATE_API_KEY
                  valueFrom:
                    secretKeyRef:
                      key: TWINGATE_API_KEY
                      name: RELEASE-NAME-twingate-operator
                - name: TWINGATE_NETWORK
                  value: <network slug>
                - name: TWINGATE_HOST
                  value: twingate.com
                - name: TWINGATE_DEFAULT_RESOURCE_TAGS
                  value: '{"cluster":"test-cluster","owner":"eran"}'
                - name: TWINGATE_REMOTE_NETWORK_ID
                  value: <remote network id>
                - name: FOO
                  value: bar
              image: twingate/kubernetes-operator:latest
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8080
              name: twingate-operator
              readinessProbe:
                httpGet:
                  path: /healthz
                  port: 8080
              resources:
                limits:
                  cpu: 100m
                  memory: 128Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
          imagePullSecrets:
            - name: my-pull-secret
          nodeSelector:
            foo: bar
          priorityClassName: myPriorityClass
          securityContext:
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-twingate-operator
          tolerations:
            - foo: bar
